packages/story-editor/src/components/panels/design/videoSegment/videoSegment.js

```diff
@@ -0,0 +1,209 @@
+ */
+import PropTypes from 'prop-types';
+import { __ } from '@googleforcreators/i18n';
+import styled from 'styled-components';
+import {
+  Text,
+  Button,
+  THEME_CONSTANTS,
+  BUTTON_SIZES,
+  BUTTON_TYPES,
+  BUTTON_VARIANTS,
+  useSnackbar,
+  Slider,
+} from '@googleforcreators/design-system';
+import { useCallback, useState } from '@googleforcreators/react';
+import { createPage, ELEMENT_TYPES } from '@googleforcreators/elements';
+import { useFeature } from 'flagged';
+
+/**
+ * Internal dependencies
+ */
+import { Row as DefaultRow } from '../../../form';
+import { SimplePanel } from '../../panel';
+import { getCommonValue } from '../../shared';
+import { useLocalMedia } from '../../../../app/media';
+import { useStory } from '../../../../app/story';
+import useInsertElement from '../../../canvas/useInsertElement';
+
+const Row = styled(DefaultRow)`
+  margin-top: 2px;
+`;
+
+const StyledButton = styled(Button)`
+  padding: 12px 8px;
+`;
+
+const StyledSlider = styled(Slider)`
+  width: 100%;
+`;
+
+const HelperText = styled(Text).attrs({
+  size: THEME_CONSTANTS.TYPOGRAPHY.PRESET_SIZES.X_SMALL,
+})`
+  color: ${({ theme }) => theme.colors.fg.tertiary};
+`;
+
+function VideoSegmentPanel({ pushUpdate, selectedElements }) {
+  const MIN_SEGMENT_LENGTH = 10;
+  const MAX_SEGMENT_LENGTH = 60;
+  const resource = getCommonValue(selectedElements, 'resource');
+  const elementId = getCommonValue(selectedElements, 'id');
+  const enableSegmentVideo = useFeature('segmentVideo');
+
+  const onChangeSegmentTime = (value) => {
+    pushUpdate({ segmentTime: value }, true);
+  };
+
+  const segmentTime = getCommonValue(selectedElements, 'segmentTime', 20);
+  const insertElement = useInsertElement();
+
+  const { pages, currentPage, addPageAt, deleteElementById } = useStory(
+    ({
+      state: { pages, currentPage },
+      actions: { addPageAt, deleteElementById },
+    }) => ({
+      pages,
+      currentPage,
+      addPageAt,
+      deleteElementById,
+    })
+  );
+
+  const { segmentVideo } = useLocalMedia(({ actions: { segmentVideo } }) => ({
+    segmentVideo,
+  }));
+
+  const [isSegmenting, setIsSegmenting] = useState();
+  const showSnackbar = useSnackbar(({ showSnackbar }) => showSnackbar);
+
+  const segmentButtonText = isSegmenting
+    ? __('Segmenting…', 'web-stories')
+    : __('Segment', 'web-stories');
+
+  const handleSegmentation = useCallback(async () => {
+    let initialized = false;
+    let processedFiles = 0;
+    setIsSegmenting(true);
+    showSnackbar({
+      message: __('Video segmentation in progress', 'web-stories'),
+      dismissible: true,
+    });
+
+    const originalElementId = elementId;
+    const pageIds = pages.map(({ id }) => id);
+    const originalPageIndex = pageIds.indexOf(currentPage.id);
+
+    await segmentVideo({ resource, segmentTime }, (segmentedFiles) => {
+      // return the "addElement" function
+      // called via "onUploadSuccess" of the segmented file
+      return (element) => {
```

Comment:
NB: this function (which is an `onUploadSuccess` handler) receives `{ id, resource}`, not an "element".